src/sentry/api/endpoints/release_deploys.py

```diff
@@ -0,0 +1,129 @@
+
+from sentry.api.base import DocSection
+from sentry.api.bases.organization import OrganizationReleasesBaseEndpoint
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.paginator import OffsetPaginator
+from sentry.api.serializers import serialize
+from sentry.models import Deploy, Environment, Release
+
+
+class DeploySerializer(serializers.Serializer):
+    name = serializers.CharField(max_length=64, required=False)
+    environment = serializers.CharField(max_length=64)
+    url = serializers.URLField(required=False)
+    dateStarted = serializers.DateTimeField(required=False)
+    dateFinished = serializers.DateTimeField(required=False)
+
+
+class ReleaseDeploysEndpoint(OrganizationReleasesBaseEndpoint):
+    doc_section = DocSection.RELEASES
+
+    def get(self, request, organization, version):
+        """
+        List a Release's Deploys
+        ````````````````````````
+        Return a list of deploys for a given release.
+        :pparam string organization_slug: the organization short name
+        :pparam string version: the version identifier of the release.
+
+        """
+        try:
+            release = Release.objects.get(
+                version=version,
+                organization=organization
+            )
+        except Release.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        if not self.has_release_permission(request, organization, release):
+            raise PermissionDenied
+
+        queryset = Deploy.objects.filter(
+            organization_id=organization.id,
+            release=release
+        )
+
+        return self.paginate(
+            request=request,
+            queryset=queryset,
+            order_by='-date_finished',
+            paginator_cls=OffsetPaginator,
+            on_results=lambda x: serialize(x, request.user),
+        )
+
+    def post(self, request, organization, version):
+        """
+        List a Release's Deploys
+        ````````````````````````
+        Return a list of deploys for a given release.
+        :pparam string organization_slug: the organization short name
+        :pparam string version: the version identifier of the release.
+
+        """
+        try:
+            release = Release.objects.get(
+                version=version,
+                organization=organization
+            )
+        except Release.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        if not self.has_release_permission(request, organization, release):
+            raise PermissionDenied
+
+        serializer = DeploySerializer(data=request.DATA)
+
+        if serializer.is_valid():
+            result = serializer.object
+            try:
+                env = Environment.objects.get(
+                    organization_id=organization.id,
+                    name=result['environment']
+                )
+            except Environment.DoesNotExist:
+                # TODO(jess) should we be creating
```

Comment:
should we be creating environments here? my assumption is yes, because the only other place we create them is events, and the events can't happen until after the deploy..